docs/sync-overview.md

```diff
@@ -0,0 +1,57 @@
+# Sync Overview
+
+This document provides a high-level overview of how syncing works.  **Note**: each component has its own quirks and will handle sync slightly differently than the general process described here.
+
+## General flow and architecture
+
+- **Crates involved**:
+  - The `sync15` and `support/sync15-traits` handle the general syncing logic and define the `SyncEngine` trait
+  - `SyncEngine` is implemented by each component crate (login, places, autofill, etc).
+  - `sync_manager` manages the overall syncing process.
+- **High level sync flow**:
+  - Sync is initiated by the application that embeds application-services.
+  - The application calls `SyncManager.sync()` to start the sync process.
+  - `SyncManager` creates `SyncEngine` instances to sync the individual components
+
+### Sync manager
+
+[`SyncManager`](https://github.com/mozilla/application-services/blob/main/components/sync_manager/src/manager.rs) is responsible for performing the high-level parts of the sync process:
+  - The consumer code calls it's `sync()` function to start the sync, passing
```

Comment:
Can we link to the `SyncParams` somewhere? I think it would be useful to see what passes into the sync function